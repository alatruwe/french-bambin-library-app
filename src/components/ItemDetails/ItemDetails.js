import React from "react";

export default class ItemDetails extends React.Component {
  static defaultProps = {};

  // add function handle click event "send an email"

  render() {
    const { title, description, image } = this.props;
    return (
      <div>
        <div>
          <h2>{title}</h2>
          <img src={image} width="250" height="250" />
          <p>{description}</p>
        </div>
        <button
          type="button"
          //onClick={this.handleSendEmail}
        >
          Send an request
        </button>
      </div>
    );
  }
}
