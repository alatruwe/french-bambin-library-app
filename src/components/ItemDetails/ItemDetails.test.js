import React from "react";
import ReactDOM from "react-dom";
import ItemDetails from "./ItemDetails";
import renderer from "react-test-renderer";

describe(`ItemDetails component`, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ItemDetails />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<ItemDetails />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
