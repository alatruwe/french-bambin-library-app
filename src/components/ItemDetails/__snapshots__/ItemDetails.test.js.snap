// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails component renders the UI as expected 1`] = `
<div>
  <div>
    <h2 />
    <img
      height="250"
      width="250"
    />
    <p />
  </div>
  <button
    type="button"
  >
    Send an request
  </button>
</div>
`;
